### ISSUE TITLE
Python 3.8 support

### ISSUE THREAD

**Issue by romainr** [python-zk/kazoo#607]
I could not find an issue about it.

`async` is a reserved keyword now and will conflict, e.g. https://docs.python.org/3/reference/compound_stmts.html#async-def

```
python3.8/site-packages/kazoo/recipe/partitioner.py", line 185
    self._child_watching(self._allocate_transition, async=True)
                                                    ^
SyntaxError: invalid syntax
```

**Comment by romainr** [comment]
For the record, I was on kazoo 2.0 and 2.7 compiles without reserved keywords.

I did not test the functionalities at run time yet.

**Comment by StephenSorriaux** [comment]
Hello,

Kazoo officially supports Python 3.8. The current Kazoo version (2.7.0) would display a warning that was fixed in #609 and it will be integrated in the next Kazoo release.

**Comment by romainr** [comment]
Great to know!

Adding 3.8 to https://pypi.org/project/kazoo/ would document it well
